nodebuilder/share/share.go

```diff
@@ -23,6 +25,44 @@ type GetRangeResult struct {
 	Proof  *types.ShareProof
 }
 
+// MarshalJSON marshals an GetRangeResult to JSON. Uses tendermint encoder for proof for compatibility.
+func (getRangeResult *GetRangeResult) MarshalJSON() ([]byte, error) {
+	type Alias GetRangeResult
+	proof, err := tmjson.Marshal(getRangeResult.Proof)
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(&struct {
+		Proof json.RawMessage `json:"Proof"`
+		*Alias
+	}{
+		Proof: proof,
+		Alias: (*Alias)(getRangeResult),
+	})
+}
+
+// UnmarshalJSON unmarshals an GetRangeResult from JSON. Uses tendermint decoder for proof for compatibility.
+func (getRangeResult *GetRangeResult) UnmarshalJSON(data []byte) error {
+	type Alias GetRangeResult
+	aux := &struct {
+		Proof json.RawMessage `json:"Proof"`
+		*Alias
+	}{
+		Alias: (*Alias)(getRangeResult),
+	}
+	if err := json.Unmarshal(data, &aux); err != nil {
```

Comment:
I this necessary? Can't we go the same approach as with ExtendedHeader where we use `tmjson` only?

Reply:
switched to marshaling whole type